### DISCUSSION TITLE
Configure automerge in regex capture group

### DISCUSSION THREAD

**Discussion by TheKangaroo** [renovatebot/renovate#27804] [Request Help] [answered]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab / 

### Please tell us more about your question or problem

We run renovate daily and use the automerge feature a lot.
Because of automerge limitations (only one platform automerge in GitLab PR mode is possible per run) we started to group all our automergable dependencies into one "automergable deps" group like so:
```
    // automerge config
    {
      groupName: "automergeable deps",
      matchUpdateTypes: ["digest"],
      automerge: true,
    },
    {
      groupName: "automergeable deps",
      matchManagers: ["pre-commit", "gitlabci-include", "terraform"],
      matchUpdateTypes: ["minor", "patch"],
      automerge: true,
    },
...
```
This works well for the global configuration but I would like to add some dependencies from our regex manager (right inside the target file) to this automerge group. I couldnt figure out how (or if) it is possible to configure automerge or adding to group with the regex manager capture groups.

I tried the following config:
```
# renovate regex manager config (note the optional automerge capturegroup):
  customManagers: [
    {
      customType: "regex",
      fileMatch: ["\\.y[a]?ml$"],
      matchStrings: [
        '# renovate: datasource=(?<datasource>.*?) depName=(?<depName>.*?)( automerge=(?<automerge>.*?))?\\s.*?[=|:]\\s*"?(?<currentValue>[^"]+?)"?\\s',
      ],
    },
  ],

# target yaml
---
# renovate: datasource=gitlab-tags depName=my/project automerge=true
app_version: "1.0.0"
```
But this does not seem to work.

Is there any way how I can achive this?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
You can do this with a packageRules entry. You match against the dependencies you want to add and then give them the same groupName as the group you want to add to. In other words, multiple packageRules can add to the same groupName

**Reply by TheKangaroo** [reply]
Thanks, I already use packageRules, my first codeblock was an example of my packageRules where I do exactly this, i.e. group all sorts of things I want to automerge into the same group. I didn't mention this, though. 
The dependencies I want to add now come from different datasources and dependency names. The only thing they have in common is that I use the regex manager with the above configuration (but I also use this manager for dependencies I do not want to automerge).
So I thought it might be possible to add the automerge via an "annotation" in the regex match string. However, what I tested in the example above didn't work.

**Reply by rarkins** [reply, marked as answer]
No, it's not possible. We don't want to overload regex managers with every possible configuration option (e.g. groupName, automerge, labels, assignees, reviewers, etc) because it's too long of a list. You need another packageRules entry, this is what packageRules are for. i.e. regex manager is for extracting dependency information, packageRules is for deciding what to do about them (grouping, automerging, etc)

**Reply by TheKangaroo** [reply]
Thanks for the clarification. It's good to know that this is just out of scope and I haven't done anything wrong in my configuration :)
